arch/x86/pagetables.c

```diff
@@ -92,6 +92,17 @@ static inline void dump_pte(void *entry, mfn_t table, int level, int index) {
            level, index, paddr, flags);
 }
 
+#if defined(__x86_64__)
+static inline int canonical_va(void *va) {
+    int sign = _ul(va) >> 47 & 1;
+    return sign * 0xfffful == _ul(va) >> 48;
```

Comment:
The 48 bits (magic number BTW) is correct for 4-Level page tables, but becomes incorrect for 5-level ones. Maybe add detection of the 5-level support and the enablement to distinguish between the two cases?

Reply:
It does not have to be implemented with this commit. Maybe create an issue for handling the 5-level pt instead. And here just replace the magic numbers with some variable in boot_flags for example.